### ISSUE TITLE
Linked fiters won't work if you use a question to populate filter values

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#31218]
### Describe the bug

Let's say you build 2 questions
1) select * from people (save it, get the id)
2) select * from {{#ID of the previous question}} as source

now add the people table in a dashboard and add 2 filters
1) first filter get the values from the first question
2) second filter get the values from the second question, and link it with the first question.

It "should" work, as one question is getting the values from the first one, but it doesn't

Let's try with GUI questions!... doesn't work either

### To Reproduce

Above

### Expected behavior

We should be able get linked filter values, at least when we use the GUI questions (we have all the metadata to make it work)

### Logs

If you use GUI questions you get
	SELECT "source"."city" AS "city" FROM (SELECT "source"."id" AS "id", "source"."address" AS "address", "source"."email" AS "email", "source"."password" AS "password", "source"."name" AS "name", "source"."city" AS "city", "source"."longitude" AS "longitude", "source"."state" AS "state", "source"."source" AS "source", "source"."birth_date" AS "birth_date", "source"."zip" AS "zip", "source"."latitude" AS "latitude", "source"."created_at" AS "created_at" FROM (SELECT "public"."people"."id" AS "id", "public"."people"."address" AS "address", "public"."people"."email" AS "email", "public"."people"."password" AS "password", "public"."people"."name" AS "name", "public"."people"."city" AS "city", "public"."people"."longitude" AS "longitude", "public"."people"."state" AS "state", "public"."people"."source" AS "source", "public"."people"."birth_date" AS "birth_date", "public"."people"."zip" AS "zip", "public"."people"."latitude" AS "latitude", "public"."people"."created_at" AS "created_at" FROM "public"."people") AS "source") AS "source" WHERE ("source"."city" IS NOT NULL) AND (("source"."city" <> $1) OR ("source"."city" IS NULL)) GROUP BY "source"."city" ORDER BY "source"."city" ASC LIMIT 1000
2023-05-31 17:38:59.949 GMT [70] DETAIL:  parameters: $1 = ''

so no filtering happenning there

### Information about your Metabase installation

```JSON
v1.46.4
```

### Severity

P2?

**Comment by calherries** [comment]
Repro'd, this happens if you use linked filters on questions with nested sources. It's easier to understand with GUI questions, because you don't need to add field filter template tags to the query. Here were my repro steps:

- 1. Create GUI question: select * from people. save it, name it "people question")
- 2. Create GUI question: select * from "people question". save it, name it "nested people question")
- 3. Now add the second question to a dashboard
- 4. Add a Text filter. Set it to filter values on the "nested people question" dashcard's "Source" column.
- 5. Add a second Text filter. Set it to filter values on the "nested people question" dashcard's "Name" column. 
- 6. Link the first filter with the second.
- 7. Select a value for the second Text filter (choose any name), and observe that the filter options are not limited at all.

Try steps 3-7 except with the first GUI question, and observe that the filter options for the second Text filter are limited.

**Comment by qnkhuat** [comment]
@calherries by "Link the first filter with the second", do you mean to go to the first filter, toggle the linked filter with the 2nd filter, or vice versa?

**Comment by qnkhuat** [comment]
I reproed this but instead of link the 2nd filter to the 1st, I did the opposite.

and then when I select a value for the 2nd filter, the 1st is narrowed down.

I think this is the correct behavior, check this doc for linked filter
> If you toggle on one of these dashboard filters, selecting a value for that filter will limit the available choices for this filter.

**Comment by calherries** [comment]
Yeah, I couldn't reproduce this anymore. 

Here were the steps I tried:

- 1. Create GUI question: select * from people. save it, name it "people question")
- 2. Create GUI question: select * from "people question". save it, name it "nested people question")
- 3. Now add the first and second question to a dashboard
- 4. Add a Text filter. Set it to filter values on the "people question" dashcard's "Source" column.
- 5. Add a second Text filter. Set it to filter values on the "nested people question" dashcard's "Name" column. 
- 6. Now limit the first filter's choices by the second filter.
- 7. Select a value for the first Text filter (choose any name), and observe that the filter options are not limited at all.

I also tried it the other way around, i.e. limiting the second filter's choices by the first filter. Both work on v1.46.4
